### PR TITLE
Fix `is_query` function

### PR DESCRIPTION
**Pull request by swtwsk**

### CODE REVIEW

File: streaming_jupyter_integrations/magics.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def flink_execute_sql(self, line: str, cell: str) -> None:
     @__interrupt_signal_decorator
     def __flink_execute_sql_internal(self, stmt: str) -> None:
         task = self.__internal_execute_sql(stmt)
-        if is_dml(stmt) or is_query(stmt):
```
Comment (niladrem) [inline streaming_jupyter_integrations/magics.py:139]: why did you remove is_query here?

Comment (swtwsk) [reply streaming_jupyter_integrations/magics.py:139]: In the previous version of the code `SELECT` was not considered part of DML. In this change all DQLs (so not only `SELECT`, but `DESCRIBE` and `EXPLAIN` too) are considered a subset of DMLs, so there's no point in checking it twice.
`is_query` is still a think in other magic though, as `INSERT` is a DML-that's-not-query and it is not returning anything to show.

Comment (swtwsk) [reply streaming_jupyter_integrations/magics.py:139]: In the previous version of the code `SELECT` was not considered part of DML. In this change all DQLs (so not only `SELECT`, but `DESCRIBE` and `EXPLAIN` too) are considered a subset of DMLs, so there's no point in checking it twice.
`is_query` is still a thing in other magic though, as `INSERT` is a DML-that's-not-query and it is not returning anything to show.

Of course, we can have a conversation whether DQLs should be considered a subset of DML at all. IMO it can be — it does not hurt us right now and is semantically correct.

File: tests/test_sql_utils.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+from streaming_jupyter_integrations.sql_utils import is_ddl, is_dml, is_query
+
+
+# Some of the following examples are taken from Flink SQL documentation
+# https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/overview/
+class TestSqlUtils(unittest.TestCase):
+    def test_simple_select(self):
+        simple_select = "   SELECT   * FROM\n\n  myTable\n\t WHERE   someField\r = \r\nsomeOtherField \n\n "
+        assert not is_ddl(simple_select)
```
Comment (maver1ck) [inline tests/test_sql_utils.py:11]: Any reasons we're mixing `assert` with `assertTrue` in the tests ?

Comment (swtwsk) [reply tests/test_sql_utils.py:11]: I've started with plain `assert` as it is what's in other tests. However when I decided to use `with self.subTest` I started using `assertTrue` in those clauses. Probably should be changed to `assertTrue`/`assertFalse` everywhere

File: streaming_jupyter_integrations/sql_utils.py [hunk 3]
```diff
@@ -1,25 +1,71 @@
...
-    ]
-    for dml in valid_dml_statements:
-        if dml in sql.lower():
-            return True
-    return False
+    if not sql or not sql.strip():
+        return False
+
+    parsed = sqlparse.parse(sql.upper())[0]
+    (first_token_idx, first_token) = parsed.token_next(-1)
```
[the remark this answers is not in the document]
Comment (swtwsk) [reply streaming_jupyter_integrations/sql_utils.py:51]: You're right, an habit from other languages 😅

File: streaming_jupyter_integrations/sql_utils.py [hunk 4]
```diff
@@ -1,25 +1,71 @@
...
+    'CREATE',
+    'DROP',
+    'ALTER',
+    'USE',
+    'LOAD',
+    'UNLOAD'
+}
+
+QUERY_KEYWORDS = {
+    'VALUES',
```
Comment (maver1ck) [inline streaming_jupyter_integrations/sql_utils.py:15]: Is there any example of query starting with VALUES ?

Comment (swtwsk) [reply streaming_jupyter_integrations/sql_utils.py:15]: According to the grammar from the docs ( https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/overview/#syntax ), query starts with `VALUES`, `WITH`, or `SELECT`. Of course, it's a response for a more general "what is a query?" question (as we can do `INSERT INTO table VALUES (...)` instead of `INSERT INTO table (SELECT ...)`) and simple `VALUES (1, 2, 3)` does not make sense.

... however, Flink does not complain about this. Just does not do anything with it:
obraz

File: streaming_jupyter_integrations/magics.py (hunk 1 above)
Comment (swtwsk) [reply streaming_jupyter_integrations/magics.py:139]: There is a difference between `is_dml` and `is_dql` now :)

File: streaming_jupyter_integrations/sql_utils.py [hunk 5]
```diff
@@ -1,25 +1,70 @@
...
+    *QUERY_KEYWORDS,
+    'DESCRIBE',
+    'DESC',
+    'EXPLAIN',
+    'SHOW'
+}
+
+DML_PREFIXES: Dict[str, List[str]] = {
+    'INSERT': [],
+    'EXECUTE': ['INSERT']
```
Comment (maver1ck) [inline streaming_jupyter_integrations/sql_utils.py:30]: I'm still thinking if this part is not over-engineered.
Is there any use of `EXECUTE` in Flink other than `EXECUTE INSERT`?

Comment (swtwsk) [reply streaming_jupyter_integrations/sql_utils.py:30]: Afaik no, there is no such place. It's more about tokenization — I imagine someone would write `EXECUTE   INSERT` or `EXECUTE\n  INSERT` or something like that, and so we cannot rely on simple `sql.starts_with("EXECUTE INSERT")`. So we can either write a whitespace-agnostic regex to match against, or use `sqlparse` and its tokenizator to check for the proper start. Right now it relies on the latter.
As I started to think about that, maybe I can change `DML_PREFIXES` from `Dict[str, List[str]]` to just a `List[str]` — it would be more "generic" to try to recursively walk and try to match every subsequent token 🙂

Comment (swtwsk) [reply streaming_jupyter_integrations/sql_utils.py]: Finally changed it to a plain regex.

**Review by niladrem — APPROVED** [review]
